airbyte-cdk/python/airbyte_cdk/sources/file_based/default_file_based_availability_strategy.py

```diff
@@ -72,11 +73,16 @@ def _check_parse_record(self, stream: AbstractFileBasedStream, file: RemoteFile)
         except Exception as exc:
             raise CheckAvailabilityError(FileBasedSourceError.ERROR_READING_FILE, stream=stream.name, file=file.uri) from exc
 
-        if stream.config.input_schema:
-            if not stream.record_passes_validation_policy(record):
+        if stream.config.input_schema and not _is_read_request(stream):
+            # On read requests, we don't bother checking that a record conforms to the schema
+            # because it will be checked when we're reading the record
+            if not conforms_to_schema(record, stream.config.input_schema):
                 raise CheckAvailabilityError(
                     FileBasedSourceError.ERROR_VALIDATING_RECORD,
                     stream=stream.name,
                     file=file.uri,
-                    validation_policy=stream.config.validation_policy,
                 )
+
+
+def _is_read_request(stream: AbstractFileBasedStream) -> bool:
```

Comment:
Without this, we end up skipping the stream during `read` because it doesn't pass the availability check. I don't love the way that I'm handling it here though so am open to suggestions.

Reply:
it's not super clear to me why we don't want to verify the schema. Is it to allow us to process the file even if the first record is not conforming?

~I think it's worth adding a comment explaining why the presence of the catalog_schema means we're doing a read~
Can we move this method to AbstractFileBasedStream? The fact that the presence of a catalog_schema means we're reading seems like an implementation detail

Reply:
The interface's doc says
> If boolean is true, then the stream is available, and no str is required. Otherwise, the stream is unavailable for some reason and the str should describe what went wrong and how to resolve the unavailability, if possible.

Can we use a different object to discard individual files during the read?

Reply:
To summarize the offline sync: for the `emit_records` and `skip_records` validation policies, we're not going to check a record against the input schema, even during the initial `check`, because if users have those options configured their syncs can continue even if the schema changes. However, on the `wait` policy, we will check that a record conforms to the schema during the stream availability check; if it fails, we won't start syncing that stream (given that a record didn't conform to the schema, the sync would have been halted anyway).

Regarding the `wait` case, I felt that it was more consistent to check the record regardless of whether the user input a schema, or we were using the schema from the catalog. The behavior should be consistent between these two cases.